airflow/models/taskinstance.py

```diff
@@ -877,6 +877,19 @@ def refresh_from_db(self, session: Session = NEW_SESSION, lock_for_update: bool
         else:
             self.state = None
 
+    @classmethod
+    def from_ti_key(cls, ti_key, session) -> TaskInstance | None:
```

Comment:
This method sounds like creating a ti from ti key; I’d name this something like `get_from_key`